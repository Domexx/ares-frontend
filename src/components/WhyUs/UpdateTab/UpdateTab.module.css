/* === === === === WhyUs/UpdateTab === === === === */

.UpdateTab {
  display: flex;
  flex: 1;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-top: 50px;
}

.UpdateTab-item {
  display: flex;
  flex-basis: 46%;
  flex-direction: column;
  margin-bottom: 50px;
  border-radius: 12px;
  background: var(--clr-dark-500);
}

.UpdateTab-itemContent {
  display: block;
  height: 100%;
  max-height: 584px;
}

.UpdateTab-head {
  display: flex;
  align-items: center;
  padding: 28px;
}

.UpdateTab-headAvatar {
  overflow: hidden;
  width: 72px;
  height: 72px;
  margin-right: 10px;
  border-radius: 50%;
  background: var(--clr-dark-400);
}

.UpdateTab-headText-update {
  color: var(--clr-text-primary);
}

.UpdateTab-text {
  padding: 28px;
}

.UpdateTab-foot {
  padding: 28px;
}

.UpdateTab-tag {
  margin-right: 10px;
  padding: 12px 28px 12px 28px;
  border-radius: 12px;
  font-size: 12px;
  background: var(--clr-dark-300);
}

@media (--phone) {
  .UpdateTab {
    display: flex;
    flex: 1;
    flex-direction: column;
    flex-wrap: wrap;
    margin-top: 50px;
    padding: 0 40px 0 40px;
  }

  .UpdateTab-headText-update {
    font-size: 14px;
  }

  .UpdateTab-headText-date {
    font-size: 12px;
  }

  .UpdateTab-text {
    font-size: 14px;
  }
}
