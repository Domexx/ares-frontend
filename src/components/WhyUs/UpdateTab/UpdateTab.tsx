import React from 'react';
import Image from 'next/image';
import useTranslation from 'next-translate/useTranslation';
import { Tag, Updates, UpdatesItem } from '../../../shared/interfaces/WhyUsItem';
import s from './UpdateTab.module.css';

export type Props = {
  main: Updates;
};

const UpdateTab: React.FC<Props> = ({ main }) => {
  const { t } = useTranslation('maintenance');

  const items = main.items.map((item: UpdatesItem, index: number) => {
    return (
      // eslint-disable-next-line react/no-array-index-key
      <div key={index} className={s['UpdateTab-item']}>
        <div className={s['UpdateTab-head']}>
          <div className={s['UpdateTab-headAvatar']}>
            <Image src={item.author_look} layout="fixed" width={64} height={110} alt={item.author} />
          </div>
          <div className={s['UpdateTab-headText']}>
            <p className={s['UpdateTab-headText-update']}>
              {item.author} {t('updated_posted')}
            </p>
            <span className={s['UpdateTab-headText-date']}>
              {t('updated_date')} {item.date}
            </span>
          </div>
        </div>
        <p className={s['UpdateTab-text']}>{item.update_text}</p>
        <div className={s['UpdateTab-itemContent']}>
          <Image src={item.image} layout="responsive" height={333} width={584} objectFit="contain" alt={item.author} />
        </div>
        <div className={s['UpdateTab-foot']}>
          {item.tags.map((tag: Tag, key: number) => {
            return (
              // eslint-disable-next-line react/no-array-index-key
              <span key={key} className={s['UpdateTab-tag']}>
                {tag.name}
              </span>
            );
          })}
        </div>
      </div>
    );
  });
  return <div className={s['UpdateTab']}>{items}</div>;
};

export default UpdateTab;
