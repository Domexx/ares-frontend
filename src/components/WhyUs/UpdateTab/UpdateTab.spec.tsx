import React from 'react';

import { render } from '@testing-library/react';

import UpdateTab from './UpdateTab';

describe('Update Tab Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <UpdateTab
        main={{
          title: 'Updates',
          items: [
            {
              author: 'Virtual',
              author_look:
                'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
              date: '08.05.2021',
              update_text:
                'Putz dich fein heraus um dann elegant über den Habbo-Laufsteg zu gleiten.Das wird ein Spass!',
              image: '/static/images/Maintenance/Update.png',
              tags: [
                {
                  name: 'Client',
                },
                {
                  name: 'Event',
                },
              ],
            },
            {
              author: 'Virtual',
              author_look:
                'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
              date: '08.05.2021',
              update_text:
                'Putz dich fein heraus um dann elegant über den Habbo-Laufsteg zu gleiten.Das wird ein Spass!',
              image: '/static/images/Maintenance/Update.png',
              tags: [
                {
                  name: 'Client',
                },
                {
                  name: 'Event',
                },
              ],
            },
            {
              author: 'Virtual',
              author_look:
                'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
              date: '08.05.2021',
              update_text:
                'Putz dich fein heraus um dann elegant über den Habbo-Laufsteg zu gleiten.Das wird ein Spass!',
              image: '/static/images/Maintenance/Update.png',
              tags: [
                {
                  name: 'Client',
                },
                {
                  name: 'Event',
                },
              ],
            },
          ],
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
