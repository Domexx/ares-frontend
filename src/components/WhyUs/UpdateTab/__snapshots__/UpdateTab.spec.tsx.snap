// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Tab Component should render properly 1`] = `
<div
  class="UpdateTab"
>
  <div
    class="UpdateTab-item"
  >
    <div
      class="UpdateTab-head"
    >
      <div
        class="UpdateTab-headAvatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="UpdateTab-headText"
      >
        <p
          class="UpdateTab-headText-update"
        >
          Virtual
           
          updated_posted
        </p>
        <span
          class="UpdateTab-headText-date"
        >
          updated_date
           
          08.05.2021
        </span>
      </div>
    </div>
    <p
      class="UpdateTab-text"
    >
      Putz dich fein heraus um dann elegant über den Habbo-Laufsteg zu gleiten.Das wird ein Spass!
    </p>
    <div
      class="UpdateTab-itemContent"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 57.02054794520548% 0px 0px 0px;"
        />
        <img
          alt="Virtual"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="UpdateTab-foot"
    >
      <span
        class="UpdateTab-tag"
      >
        Client
      </span>
      <span
        class="UpdateTab-tag"
      >
        Event
      </span>
    </div>
  </div>
  <div
    class="UpdateTab-item"
  >
    <div
      class="UpdateTab-head"
    >
      <div
        class="UpdateTab-headAvatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="UpdateTab-headText"
      >
        <p
          class="UpdateTab-headText-update"
        >
          Virtual
           
          updated_posted
        </p>
        <span
          class="UpdateTab-headText-date"
        >
          updated_date
           
          08.05.2021
        </span>
      </div>
    </div>
    <p
      class="UpdateTab-text"
    >
      Putz dich fein heraus um dann elegant über den Habbo-Laufsteg zu gleiten.Das wird ein Spass!
    </p>
    <div
      class="UpdateTab-itemContent"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 57.02054794520548% 0px 0px 0px;"
        />
        <img
          alt="Virtual"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="UpdateTab-foot"
    >
      <span
        class="UpdateTab-tag"
      >
        Client
      </span>
      <span
        class="UpdateTab-tag"
      >
        Event
      </span>
    </div>
  </div>
  <div
    class="UpdateTab-item"
  >
    <div
      class="UpdateTab-head"
    >
      <div
        class="UpdateTab-headAvatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="UpdateTab-headText"
      >
        <p
          class="UpdateTab-headText-update"
        >
          Virtual
           
          updated_posted
        </p>
        <span
          class="UpdateTab-headText-date"
        >
          updated_date
           
          08.05.2021
        </span>
      </div>
    </div>
    <p
      class="UpdateTab-text"
    >
      Putz dich fein heraus um dann elegant über den Habbo-Laufsteg zu gleiten.Das wird ein Spass!
    </p>
    <div
      class="UpdateTab-itemContent"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 57.02054794520548% 0px 0px 0px;"
        />
        <img
          alt="Virtual"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="UpdateTab-foot"
    >
      <span
        class="UpdateTab-tag"
      >
        Client
      </span>
      <span
        class="UpdateTab-tag"
      >
        Event
      </span>
    </div>
  </div>
</div>
`;
